src/Avalonia.Controls/TextBox.cs

```diff
@@ -691,7 +695,7 @@ private int CoerceCaretIndex(int value, int length)
             {
                 return length;
             }
-            else if (value > 0 && text[value - 1] == '\r' && text[value] == '\n')
+            else if (value > 0 && text[value - 1] == '\r' && value < length && text[value] == '\n')
```

Comment:
Maybe moving the `value < length` after `value > 0` to improve little performance.

`value > 0 && value < length && text[value - 1] == '\r' && text[value] == '\n'` 

Because the code checks the number value is faster than get the index and checks the char.

Reply:
I put it there because failing the length check is very rare. It will be superseded by #1950 anyway